pkg/network/node/egressip.go

```diff
@@ -61,6 +60,45 @@ func (eip *egressIPWatcher) Start(networkInformers networkinformers.SharedInform
 	return nil
 }
 
+func (eip *egressIPWatcher) Synced() {
+	link, _, err := GetLinkDetails(eip.localIP)
+	if err != nil {
+		// shouldn't happen, but obviously there's nothing to clean up...
+		return
+	}
+	label, err := egressIPLabel(link)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Could not check for stale egress IPs: %v", err))
+		return
+	}
+	addrs, err := netlink.AddrList(link, netlink.FAMILY_V4)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Could not check for stale egress IPs: %v", err))
+		return
+	}
+
+	for _, addr := range addrs {
+		ip := addr.IP.String()
+		if addr.Label == label && eip.iptablesMark[ip] == "" {
+			klog.Infof("Cleaning up stale egress IP %s", addr.IP.String())
+			err = netlink.AddrDel(link, &addr)
+			if err != nil {
+				utilruntime.HandleError(fmt.Errorf("Could not clean up stale egress IP: %v", err))
```

Comment:
Should also include the IP that is causing the problem.

Reply:
odd... I could have sworn I already responded to this.

Anyway, it already logged the IP a few lines earlier, so it will be clear. (Also, this is a backport so we don't want it to be different from the version in master anyway.)